### PR TITLE
Fixed Error if No Project Exists (URL)

### PR DESCRIPTION
**Pull request by mecyborg**
If someone access the URL as /username/any_project and that project doesn't exists, the application was reporting error.

Update/Fix: Added a check and redirect to dashboard if project doesn't exist.

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -129,6 +129,10 @@ def show_tree_content
 
   def show
     @images = []
+    if @project.blank?
+    redirect_to dashboard_path
```
Comment (rohitpaulk) [inline app/controllers/projects_controller.rb]: Code indentation isn't proper, and you haven't closed the `if` condition with an `end`. Moreover, this isn't the right way to approach the problem. If the project doesn't exist, we should throw the error right in `return_context` and render a 404.

You've also checked in the `hot_deploy` marker file.

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -129,6 +129,10 @@ def show_tree_content
...
+    flash[:alert] = 'The project does not exist.'
+    else
```
Comment (EmilyDirsh) [inline app/controllers/projects_controller.rb]: Small nitpick - fix the indentation of the nested statements, so it's a little more readable.

File: app/controllers/projects_controller.rb (hunk 1 above)
Comment (mecyborg) [inline app/controllers/projects_controller.rb]: I was also wondering whether to render 404 or redirect to dashboard. But, now it clear, thanks :)
I'll fix the code indentation and if statement.
hot_deploy shouldn't be there. No worries, I'll fix that too if it there.

**Comment by rohitpaulk** [comment]
If you decide to correct this PR and push again - please add a failing test.

**Comment by mecyborg** [comment]
This works fine according to me. Please verify :)

**Comment by rohitpaulk** [comment]
@mecyborg - Please add a failing test so that this error doesn't pop up again in the future.

**Comment by mecyborg** [comment]
@rohitpaulk Failing test added.

**Pull request merged**